pkg/conversion/deep_equal.go

```diff
@@ -174,6 +174,9 @@ func (e Equalities) deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool,
 
 	switch v1.Kind() {
 	case reflect.Array:
+		if v1.Len() != v2.Len() {
+			return false
+		}
```

Comment:
yes, the `{[]string{"1", "2", "3"}, []string{}, false},` case panics because you iterate both v1 and v2 with the same index, but v1 might be longer than v2, causing an out-of-bounds.